db/template_test.go

```diff
@@ -0,0 +1,131 @@
+
+	id, err := uuid.Parse(ids.templateID)
+	if err != nil {
+		t.Fail()
+	}
+	if _, err := a.tinkDb.instance.Exec("delete from template;"); err != nil {
+		t.Fail()
+	}
+	if err := a.tinkDb.CreateTemplate(a.ctx.context, "foo", a.templateData, id); err != nil {
+		t.Logf("%s", a.templateData)
+		a.logger.Error(err)
+		t.Fail()
+	}
+}
+
+// TestConcurrentCreateTemplate : inserts five records concurrently and validates
+// whether or not all five records are actually committed.
+func TestConcurrentCreateTemplate(t *testing.T) {
+	var err error
+	count := 10
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+	if _, err := a.tinkDb.instance.Query("delete from template;"); err != nil {
+		t.Fail()
+	}
+	var wg sync.WaitGroup
+	for i := 0; i < count; i++ {
+		wg.Add(1)
+		id, err := uuid.NewRandom()
+		if err != nil {
+			t.Fail()
+		}
+		label := fmt.Sprintf("temp%v", i)
+		t.Logf("PUSHING %v", i)
+		if err != nil {
+			t.Fail()
+		}
+		go func(k string, v string, a *testArgs, wg *sync.WaitGroup) {
+			defer wg.Done()
+			id, err := uuid.Parse(v)
+			if err != nil {
+				t.Fail()
+			}
+			if err := a.tinkDb.CreateTemplate(a.ctx.context, k, a.templateData, id); err != nil {
+				a.logger.Error(err)
+				t.Fail()
+			}
+		}(label, id.String(), a, &wg)
+	}
+	wg.Wait()
+	var resultRole int
+	if err = a.tinkDb.instance.QueryRow("select count(*) as count from template;").Scan(&resultRole); err != nil {
+		t.Fail()
+	}
+	if resultRole != count {
+		t.Error(fmt.Errorf("Tempaltes PUSHED: %v. \n MACS RETURNED: %v", count, resultRole))
+	}
+}
+
+// TestGetTemplate : validates retrieving a workflow template.
+func TestGetTemplate(t *testing.T) {
+	TestCreateTemplate(t)
+	var err error
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	filter := make(map[string]string)
+	filter["name"] = "foo"
+	if a.tinkDb.instance, err = sql.Open("postgres", a.psqlInfo); err != nil {
+		t.Fail()
+	}
+	if _, _, err = a.tinkDb.GetTemplate(a.ctx.context, filter); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
+}
+
+// TestDeleteTemplate : validates the deletion of a template.
+func TestDeleteTemplate(t *testing.T) {
+	TestInsertIntoDB(t)
+	var err error
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	if a.tinkDb.instance, err = sql.Open("postgres", a.psqlInfo); err != nil {
+		t.Fail()
+	}
+	if err = a.tinkDb.DeleteTemplate(a.ctx.context, ids.templateID); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
+}
+
+// TestUpdateTemplate : validates updating a template.
+func TestUpdateTemplate(t *testing.T) {
+	TestCreateTemplate(t)
+	var err error
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	id, err := uuid.Parse(ids.templateID)
+	filter := make(map[string]string)
+	filter["name"] = "foo"
+	if a.tinkDb.instance, err = sql.Open("postgres", a.psqlInfo); err != nil {
+		t.Fail()
+	}
+	if err = a.tinkDb.UpdateTemplate(a.ctx.context, "foo", a.templateData, id); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
```

Comment:
this doesn't look like its testing that the update changed anything right?

Reply:
TestUpdateTemplate doesn't look like its testing that an update actually happened as expected